src/rules.spec.ts

```diff
@@ -0,0 +1,350 @@
+import { getMD5Hash } from './obfuscation';
+import { ObfuscatedOperatorType, OperatorType, Rule, matchesRule } from './rules';
+
+describe('rules', () => {
+  describe('Operators', () => {
+    it('ObfuscatedOperatorTypes should match hashed OperatorTypes', () => {
+      expect(ObfuscatedOperatorType.GTE).toBe(getMD5Hash(OperatorType.GTE));
+      expect(ObfuscatedOperatorType.LTE).toBe(getMD5Hash(OperatorType.LTE));
+      expect(ObfuscatedOperatorType.LT).toBe(getMD5Hash(OperatorType.LT));
+      expect(ObfuscatedOperatorType.ONE_OF).toBe(getMD5Hash(OperatorType.ONE_OF));
+      expect(ObfuscatedOperatorType.NOT_ONE_OF).toBe(getMD5Hash(OperatorType.NOT_ONE_OF));
+      expect(ObfuscatedOperatorType.MATCHES).toBe(getMD5Hash(OperatorType.MATCHES));
+      expect(ObfuscatedOperatorType.NOT_MATCHES).toBe(getMD5Hash(OperatorType.NOT_MATCHES));
+    });
+  });
+
+  describe('matchesRule | standard rules', () => {
+    const ruleWithEmptyConditions: Rule = {
+      conditions: [],
+    };
+    const numericRule: Rule = {
+      conditions: [
+        {
+          operator: OperatorType.GTE,
+          attribute: 'totalSales',
+          value: 10,
+        },
+        {
+          operator: OperatorType.LTE,
+          attribute: 'totalSales',
+          value: 100,
+        },
+      ],
+    };
+    const ruleWithOneOfCondition: Rule = {
+      conditions: [
+        {
+          operator: OperatorType.ONE_OF,
+          attribute: 'country',
+          value: ['Canada', 'Mexico', 'USA'],
+        },
+      ],
+    };
+
+    const ruleWithNotOneOfCondition: Rule = {
+      conditions: [
+        {
+          operator: OperatorType.NOT_ONE_OF,
+          attribute: 'country',
+          value: ['Canada', 'Mexico', 'USA'],
+        },
+      ],
+    };
+
+    const semverRule: Rule = {
+      conditions: [
+        {
+          operator: OperatorType.GTE,
+          attribute: 'version',
+          value: '1.2.5',
+        },
+        {
+          operator: OperatorType.LTE,
+          attribute: 'version',
+          value: '2.4.2',
+        },
+      ],
+    };
+    const ruleWithMatchesCondition: Rule = {
+      conditions: [
+        {
+          operator: OperatorType.MATCHES,
+          attribute: 'user_id',
+          value: '\\d+',
+        },
+      ],
+    };
+    const ruleWithNotMatchesCondition: Rule = {
+      conditions: [
+        {
+          operator: OperatorType.NOT_MATCHES,
+          attribute: 'user_id',
+          value: '[0-9]+',
+        },
+      ],
+    };
+    const subjectAttributes = {
+      totalSales: 50,
+      version: '1.5.0',
```

Comment:
Make this `1.15.0` so we know more than just lexographic comparisons are taking place.

(i.e., the string "1.15.0" is less than the string "1.2.5" but semver 1.15.0 is greater than semver 1.2.5)